pilot/pkg/networking/core/v1alpha3/listener_inbound.go

```diff
@@ -262,6 +262,14 @@ func (lb *ListenerBuilder) buildInboundChainConfigs() []inboundChainConfig {
 	chainsByPort := make(map[uint32]inboundChainConfig)
 	// No user supplied sidecar scope or the user supplied one has no ingress listeners.
 	if !lb.node.SidecarScope.HasIngressListener() {
+
+		// We should not create inbound listeners in NONE mode based on the service instances
+		// Doing so will prevent the workloads from starting as they would be listening on the same port
+		// Users are required to provide the sidecar config to define the inbound listeners
```

Comment:
What about outbound listeners? We create bind to port with [true](https://github.com/istio/istio/blob/dab79d8e213b3e0c187ffe22b7608643912f2588/pilot/pkg/networking/core/v1alpha3/listener.go#L320), would it also conflict?

Reply:
Outbound listeners bind to localhost interface (127.0.0.1). 

When the interception mode is NONE, an outbound listener is created matching the port(s) of each registered service. So, it is possible that the outbound listener tries to bind to the same port used by the application. 

IMO, the disabled interception mode requires other controls and a heavily use of the Sidecar and Envoy Filter resource.

Reply:
To me interception mode is 'None' not well tested. Should we deprecate this? @hzxuzhonghu @howardjohn  - not sure if any one uses it with these bugs?

Reply:
@ramaraochavali I totally agree that the InterceptionMode NONE is not fully tested but I think deprecating it is not a good strategy. In my company, we are designing a mesh to handle > 3Mrps using the InterceptionMode NONE.

In this case, we allow administrators to define what works best for them and support multiple use-cases.

Reply:
The reason why I was thinking of deprecating is , we have not announced it in the doc https://github.com/istio/api/blob/master/annotation/annotations.yaml#L158.
If we want to keep it, we should add more tests around that update the docs. Otherwise, we might hit surprising behaviours.

Reply:
It would be best to make it more tested and documented rather than deprecated probably.. it is fairly used.

I guess my general concern with this change is we are making a big behavioral change. So what is the impact on existing users? Is EVERYONE using this mode using explicit Sidecar.ingress config, so this doesn't really matter (the only change is to make non-Sidecar case go from NACK -> empty config?)?

Reply:
> explicit Sidecar.ingress config

Sidecar.ingress already has capture Mode and that is what we use. Not sure, if some one uses sidecarIngress config and additionally use this annotation to set to None. What is the benefit of both ?

Reply:
By adding the annotation `sidecar.istio.io/interceptionMode: NONE`, the injection webhook does not add the istio-init container that creates the interception rules. In our setup, it is not an issue because we inject the sidecar using other tools.

Setting the captureMode in the Sidecar resource, among other things, sets the bindToPort of Envoy listeners. When looking only on the perspective of traffic routing, it is possible to achieve the same setup by setting the annotation  `sidecar.istio.io/interceptionMode: NONE` or setting the captureMode in the Sidecar resource.

Given that, if we can disable the injection of the istio-init container by the webhook, I do not see any issue in setting the captureMode in the Sidecar resource and deprecating the annotation `sidecar.istio.io/interceptionMode: NONE`.